addon/webextension/background/analytics.js

```diff
@@ -0,0 +1,74 @@
+/* globals main, auth, catcher, deviceInfo, communication */
+
+window.analytics = (function () {
+  let exports = {};
+
+  let telemetryPrefKnown = false;
+  let telemetryPref;
+
+  exports.sendEvent = function (action, label, options) {
+    let eventCategory = "addon";
+    if (! telemetryPrefKnown) {
+      console.warn("sendEvent called before we were able to refresh");
+      return Promise.resolve();
+    }
+    if (! telemetryPref) {
+      console.info(`Cancelled sendEvent ${eventCategory}/${action}/${label || 'none'} ${JSON.stringify(options)}`);
+      return Promise.resolve();
+    }
+    if (typeof label == "object" && (! options)) {
+      options = label;
+      label = undefined;
+    }
+    options = options || {};
+    let di = deviceInfo();
+    return new Promise((resolve, reject) => {
+      let url = main.getBackend() + "/event";
+      let req = new XMLHttpRequest();
```

Comment:
https://github.com/mozilla-services/screenshots/issues/2261 – we on the team have avoided taking good first bugs while Outreachy applications were incoming, but now that the application period is over we can probably take these ourselves.